src/test/c_api_stress_tests.cpp

```diff
@@ -0,0 +1,441 @@
+}
+TEST_F(StressCapiConfigChanges, AddedNewPipelineDuringPredictLoad) {
+    bool performWholeConfigReload = true;
+    std::set<StatusCode> requiredLoadResults = {StatusCode::OK};  // we expect full continuity of operation
+    std::set<StatusCode> allowedLoadResults = {};
+    performStressTest(
+        &StressCapiConfigChanges::triggerCApiInferenceInALoop,
+        &StressCapiConfigChanges::addNewPipeline,
+        performWholeConfigReload,
+        requiredLoadResults,
+        allowedLoadResults);
+}
+TEST_F(StressCapiConfigChanges, RetireSpecificVersionUsedDuringPredictLoad) {
+    // we declare specific version used (1) and latest model version policy with count=1
+    // then we add version 2 causing previous default to be retired
+    SetUpConfig(stressTestPipelineOneDummyConfigSpecificVersionUsed);
+    bool performWholeConfigReload = false;
+    std::set<StatusCode> requiredLoadResults = {StatusCode::OK,
+        StatusCode::PIPELINE_DEFINITION_NOT_LOADED_YET};
+    std::set<StatusCode> allowedLoadResults = {StatusCode::MODEL_VERSION_NOT_LOADED_ANYMORE};
+    performStressTest(
+        &StressCapiConfigChanges::triggerCApiInferenceInALoop,
+        &StressCapiConfigChanges::retireSpecificVersionUsed,
+        performWholeConfigReload,
+        requiredLoadResults,
+        allowedLoadResults);
+}
+TEST_F(StressCapiConfigChanges, AddNewVersionDuringGetMetadataLoad) {
+    bool performWholeConfigReload = false;                        // we just need to have all model versions rechecked
+    std::set<StatusCode> requiredLoadResults = {StatusCode::OK};  // we expect full continuity of operation
+    std::set<StatusCode> allowedLoadResults = {};
+    performStressTest(
+        &StressCapiConfigChanges::triggerCApiGetMetadataInALoop,
+        &StressCapiConfigChanges::defaultVersionAdd,
+        performWholeConfigReload,
+        requiredLoadResults,
+        allowedLoadResults);
+}
+TEST_F(StressCapiConfigChanges, RemoveDefaultVersionDuringGetMetadataLoad) {
+    std::set<StatusCode> requiredLoadResults = {StatusCode::OK,
+        StatusCode::PIPELINE_DEFINITION_NOT_LOADED_YET};  // we hit when all config changes finish to propagate
+    std::set<StatusCode> allowedLoadResults = {};
+    // we need whole config reload since there is no other way to dispose
+    // all model versions different than removing model from config
+    bool performWholeConfigReload = true;
+    performStressTest(
+        &StressCapiConfigChanges::triggerCApiGetMetadataInALoop,
+        &StressCapiConfigChanges::defaultVersionRemove,
+        performWholeConfigReload,
+        requiredLoadResults,
+        allowedLoadResults);
+}
+TEST_F(StressCapiConfigChanges, ChangeToShapeAutoDuringGetMetadataLoad) {
+    bool performWholeConfigReload = true;
+    std::set<StatusCode> requiredLoadResults = {StatusCode::OK};  // we expect full continuity of operation
+    std::set<StatusCode> allowedLoadResults = {StatusCode::PIPELINE_DEFINITION_NOT_LOADED_YET};
```

Comment:
This is expected end status - that is after we perform config change we expect this status to continue to appear? If yes this should be requiredLoadResult